allennlp_models/pretrained.py

```diff
@@ -1,194 +1,46 @@
-from typing import Optional
+from typing import Dict
 
-from allennlp.models import load_archive
-from allennlp.predictors import Predictor, TextClassifierPredictor
+from allennlp.predictors import Predictor
 
-from allennlp_models.coref import CorefPredictor
-from allennlp_models.mc.predictors import TransformerMCPredictor
-from allennlp_models.pair_classification import DecomposableAttentionPredictor
-from allennlp_models.rc import ReadingComprehensionPredictor
-from allennlp_models.structured_prediction import (
-    SemanticRoleLabelerPredictor,
-    OpenIePredictor,
-    ConstituencyParserPredictor,
-    BiaffineDependencyParserPredictor,
-)
-from allennlp_models.tagging.predictors import SentenceTaggerPredictor
+from allennlp_models.classification import models as classification_models  # noqa: F401
+from allennlp_models.coref import models as coref_models  # noqa: F401
+from allennlp_models.generation import models as generation_models  # noqa: F401
+from allennlp_models.lm import models as lm_models  # noqa: F401
+from allennlp_models.mc import models as mc_models  # noqa: F401
+from allennlp_models.pair_classification import models as pc_models  # noqa: F401
+from allennlp_models.rc import models as rc_models  # noqa: F401
+from allennlp_models.structured_prediction import models as sp_models  # noqa: F401
+from allennlp_models.tagging import models as tagging_models  # noqa: F401
 
+from allennlp_models.common.model_card import get_pretrained_models
 
-# flake8: noqa: E501
+# TODO: rename all archive files to follow standard naming.
+# Standard: task-model.yyyy-mm-dd.tar.gz OR model-task.yyyy-mm-dd.tar.gz
 
+PRETRAINED_MODELS = get_pretrained_models()
 
-def _load_predictor(archive_file: str, predictor_name: Optional[str] = None) -> Predictor:
-    """
-    Helper to load the desired predictor from the given archive.
-    """
-    archive = load_archive(archive_file)
-    return Predictor.from_archive(archive, predictor_name)
-
-
-def bert_srl_shi_2019() -> SemanticRoleLabelerPredictor:
-    predictor = _load_predictor(
-        "https://storage.googleapis.com/allennlp-public-models/bert-base-srl-2020.07.14.tar.gz"
-    )
-    return predictor
+PRETRAINED_MODELS["glove-sst"] = {
+    "archive_file": "basic_stanford_sentiment_treebank-2020.06.09.tar.gz"
+}
 
-
-def bidirectional_attention_flow_seo_2017() -> ReadingComprehensionPredictor:
-    """
-    Reading Comprehension
-
-    Based on `BiDAF (Seo et al, 2017) <https://www.semanticscholar.org/paper/Bidirectional-Attention-Flow-for-Machine-Comprehen-Seo-Kembhavi/007ab5528b3bd310a80d553cccad4b78dc496b02>`_
-    """
-    predictor = _load_predictor(
-        "https://storage.googleapis.com/allennlp-public-models/bidaf-model-2020.03.19.tar.gz"
-    )
-    return predictor
+PRETRAINED_MODELS["roberta-sst"] = {"archive_file": "sst-roberta-large-2020.06.08.tar.gz"}
```

Comment:
For these exceptions, is the problem that the models aren't available to treat like we treat the other models?

Reply:
Yes, that's right.

Reply:
Hmm, I guess we could just call `add_pretrained_model()` in here then, to keep the lists in sync.

But I'd much rather have only one list. Keeping two if them in sync is never fun.